exercises/ruby-1-b/ruby_1_b_test.rb

```diff
@@ -1,34 +1,95 @@
 require 'minitest/autorun'
-require_relative 'ruby_1_b'
+require_relative 'ruby_1_b.rb'
 
-# Common test data version: 1.2.0 4fc1acb
-class SentenceTest < Minitest::Test
-  def test_one_word_with_one_vowel
-    assert_equal "a", word_with_most_vowels("a")
+class LibraryTest < Minitest::Test
+  def setup
+    @library = Library.new
   end
 
-  def test_one_word_with_one_vowel_and_one_consonant
-    assert_equal "at", word_with_most_vowels("at")
+  def test_find_by_empty_isbn_returns_nil
+    assert_nil @library.find("")
   end
 
-  def test_one_word_with_two_vowels_and_one_consonant
-    assert_equal "tea", word_with_most_vowels("tea")
+  def test_find_by_isbn_without_corresponding_book
+    assert_nil @library.find("123456789")
   end
 
-  def test_longest_word_is_not_word_with_most_vowels
-    assert_equal "area", word_with_most_vowels("the area of a circle")
+  def test_find_first_book
+    assert @library.find("9780191604928")
   end
 
-  def test_last_word_is_word_with_most_vowels
-    assert_equal "vegetables", word_with_most_vowels("extremely fresh vegetables")
+  def test_find_last_book
+    assert @library.find("9780451524935")
   end
 
-  def test_mixed_casing
-    assert_equal "YOUR", word_with_most_vowels("TEST YOUR MIGHT")
+  def test_book_has_title
+    assert_equal "The Adventures of Tom Sawyer", @library.find("9780191604928").title
   end
 
-  def test_ignore_word_without_vowels
-    assert_equal "great", word_with_most_vowels("great rhythm")
+  def test_book_has_author
+    assert_equal "Plato", @library.find("9780718198916").author
+  end
+
+  def test_book_has_isbn
+    assert_equal "9780974514055", @library.find("9780974514055").isbn
+  end
+
+  def test_book_has_published
+    assert_equal "1988", @library.find("9780446310789").published
+  end
+
+  def test_add_copies_with_invalid_isbn_raises_argument_error
+    assert_raises(ArgumentError) { @library.add_copies("123456789", count: 10) }
+  end
+
+  def test_add_copies_with_valid_isbn_returns_the_added_count
+    assert_equal 10, @library.add_copies("9780544003415", count: 10)
+  end
+
+  def test_borrowing_when_there_are_no_copies_returns_nil
+    assert_nil @library.borrow("9780544003415")
+  end
+
+  def test_borrowing_a_book_after_copies_have_been_added_returns_the_book
```

Comment:
It's not clear to me what the `borrow` and `return` methods should actually do.

I'd also suggest we don't have a method called return, as this is going to clash when used internally in the class.

Reply:
Name options:

- `check_out`
- `check_in`

I went through the English thesaurus I had available, and remit, lend, borrow, all had their charms, but quirks.  I think the above is probably the clearest names, avoiding the `return` that going to be problematic in Ruby.